sky/backends/backend_utils.py

```diff
@@ -733,8 +750,11 @@ def write_cluster_config(to_provision: 'resources.Resources',
     yaml_path = _get_yaml_path_from_cluster_name(cluster_name)
     old_yaml_content = None
     if os.path.exists(yaml_path):
-        with open(yaml_path, 'r') as f:
-            old_yaml_content = f.read()
+        if force_overwrite:
```

Comment:
When would be the case that the cluster didn't exist but this file exists?  If this is an exceptional case to guard against, rename it to `keep_launch_fields_in_existing_config: bool`?

Reply:
The configure yaml files may not be correctly deleted if an error happens during `sky.down`. I did find multiple yaml files in the folder `~/.sky/generated` not belonging to any existing cluster. 
Good point! Let me rename the variable.